packages/devextreme/js/__internal/grids/grid_core/data_controller/m_data_helper_mixin.ts

```diff
@@ -47,7 +47,10 @@ export const DataHelperMixin = <T extends ModuleType<Controller>>(Base: T) => cl
   }
 
   protected _initDataSource() {
-    let dataSourceOptions = SPECIFIC_DATA_SOURCE_OPTION in this ? this[SPECIFIC_DATA_SOURCE_OPTION]() : this.option('dataSource');
+    let dataSourceOptions = SPECIFIC_DATA_SOURCE_OPTION in this
+      ? (this[SPECIFIC_DATA_SOURCE_OPTION] as any)()
+      : this.option('dataSource');
+
```

Comment:
Looks like this should be part of the type:

```ts
export const DataHelperMixin = <T extends ModuleType<Controller>>(Base: T) => class DataHelperMixin extends Base {
  private _getSpecificDataSourceOption?(): any
  private _normalizeDataSource?(datasource: any): any;
  private _dataSourceOptions?(): any;
  private _dataSourceFromUrlLoadMode?(): any;
protected _initDataSource(this: this & ) {
```

```ts
let dataSourceOptions: unknown = this._getSpecificDataSourceOption?.() ?? this.option('dataSource');
...
widgetDataSourceOptions = this._dataSourceOptions?.() ?? {};
...
fromUrlLoadMode: this._dataSourceFromUrlLoadMode?.(),
...
this._dataSource = this._normalizeDataSource(this._dataSource);
```

Reply:
Maybe it's better to leave it as it is (using `as any`) to avoid changinh runtime in this PR. I would better add a next PR for removing these JS-style lines.

Reply:
This file is more like a workaround, it will be removed in future completely, so no need to keep TS good in there

Reply:
👌🏻